### DISCUSSION TITLE
Real Time Model Results

### DISCUSSION THREAD

**Discussion by drvshavva** [apache/streampipes#2459] [Q&A] [answered]
I want to upload and use my machine learning models (in ONNX or H5 format) or even my mathematical models in StreamPipes. How can I develop this?

**Comment by dominikriemer** [comment]
Hi  @drvshavva did you have a look at the Python functions? That's probably a good way to integrate your custom models and you can also play back results to StreamPipes as a data stream.

**Reply by drvshavva** [reply]
Hello, first of all, thank you very much for your prompt response. Yes, I've reviewed it, but we can't integrate them into StreamPipes' toolset, can we? If I wanted to add them as a new data processor in the drag-and-drop interface, is there a way to do this? On this page https://airegio.ems-carsa.com/nfs/programme_5/call_3/call_preparation/Implementing%20AI%20REGIO%20Data%20Pipelines%20with%20Streampipes.pdf, it's mentioned that you managed to do this with the Deeplearning4j library. Do you have a recommended method or any experiments in this regard? I would appreciate it if you could help me with this, thank you in advance.

**Reply by dominikriemer** [reply, marked as answer]
Hi @drvshavva when using the Python client, you can create a data stream which is available in StreamPipes and contains the result of a function (e.g., a model).
The implementation in the document you sent was not created by members of the StreamPipes community and the code has not been contributed (unfortunately). But you can create something similar by using the Java SDK, invlude the Deeplearning4j library and create a pipeline element which has similar configuration options as described in the deliverable.
Hope this helps!
